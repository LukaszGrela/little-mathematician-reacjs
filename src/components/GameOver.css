/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/
 
.game-over {
    padding-top: 1rem;
    text-align: left;
    width: 300px;
    margin: 0 auto;
}
.game-over .title {
    font-size: 2rem;
    margin: 1.5rem 0;
    text-align: center;
}
.game-over .content {
    display: block;
    margin: 1.5rem 0;
}
.game-over .footer {
    display: block;
    text-align: center;
    margin: 1.5rem 0;
}

.game-over .title .p1 {
    color: #388E3C;
}

.game-over .title .p2 {
    color: #8BC34A;
}

.game-over .title .p3 {
    color: #CDDC39;
}

.game-over .footer button {
    height: 75px;
    width: 75px;
    margin: 10px;
    border: none;
    background: none;
}

.game-over .footer button svg {
    fill: #fff;
    width: 50px;
    height: 50px;
    -webkit-transition: all .5s ease;
    -o-transition: all .5s ease;
    transition: all .5s ease;
}
/*
.game-over .footer button svg.icon-menu {
    fill: #E64A19;
}
.game-over .footer button svg.icon-replay {
    fill: #FF9800;
}
*/
.game-over .footer button.menu-button {
    background-color: #E64A19;
}
.game-over .footer button.replay-button {
    background-color: #FF9800;
}


.game-over .footer button:hover svg {
    -webkit-transform: rotate(1turn);
    -ms-transform: rotate(1turn);
    transform: rotate(1turn);
}