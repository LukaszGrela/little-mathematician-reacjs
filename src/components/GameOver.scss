/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

@import '../sass/_variables.scss';
.game-over {
    padding-top: 1rem;
    text-align: left;
    width: 300px;
    margin: 0 auto;
    .title {
        font-size: 2rem;
        margin: 1.5rem 0;
        text-align: center;
        .p1 {
            color: $accent-color-1;
        }
        .p2 {
            color: $accent-color-2;
        }
        .p3 {
            color: $accent-color-3;
        }
    }
    .content {
        display: block;
        margin: 1.5rem 0;
    }
    .footer {
        display: block;
        text-align: center;
        margin: 1.5rem 0;
        button {
            height: 75px;
            width: 75px;
            margin: 10px;
            border: none;
            background: none;
            svg {
                fill: $primary-color-text;
                width: 50px;
                height: 50px;
                -webkit-transition: all .5s ease;
                -o-transition: all .5s ease;
                transition: all .5s ease;
            }
            &:hover svg {
                -webkit-transform: rotate(1turn);
                -ms-transform: rotate(1turn);
                transform: rotate(1turn);
            }
            &.menu-button {
                background-color: $accent-color-6;
            }
            &.replay-button {
                background-color: $accent-color-5;
            }
        }
    }
}
