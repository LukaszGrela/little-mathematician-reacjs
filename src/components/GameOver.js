/*
   Copyright 2018 ≈Åukasz 'Severiaan' Grela GrelaDesign

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/
 
import React, { Component } from 'react';

import './GameOver.css'
import IconReplay from '../icons/IconReplay';
import IconMenu from '../icons/IconMenu';

class GameOver extends Component {
    constructor(props) {
        super(props);
        this.state = {}

        this.handleClick = this.handleClick.bind(this);
    }
    handleClick(action) {
        this.props.onAction && this.props.onAction(action);
    }
    render() {
        return (
            <div className='game-over'>
                <div className='title'><span className='p1'>Game </span><span className='p2'>Over</span><span className='p3'>!</span></div>
                <div className='content'>You have resolved {this.props.hudCorrectAnswers} equation{this.props.hudCorrectAnswers === 1 ? '' : 's'} correctly.</div>
                <div className='footer'>
                    <button
                        className='replay-button'
                        onClick={() => {
                            this.handleClick('replay')
                        }}><IconReplay /></button>
                    <button
                        className='menu-button'
                        onClick={() => {
                            this.handleClick('menu')
                        }}><IconMenu /></button>
                </div>
            </div>
        );
    }
}

export default GameOver;