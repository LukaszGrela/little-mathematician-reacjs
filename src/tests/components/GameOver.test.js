import React from 'react';
import { shallow } from 'enzyme';
import GameOver from '../../components/GameOver';

let wrapper, onAction, hudCorrectAnswers;

beforeEach(() => {
    onAction = jest.fn();
    hudCorrectAnswers = 10;
    wrapper = shallow(<GameOver
        onAction={onAction}
        hudCorrectAnswers={hudCorrectAnswers}
    />);
})

test('should render GameOver correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should render GameOver correctly with 1 correct answer', () => {
    wrapper.setProps({
        hudCorrectAnswers:1
    });
    expect(wrapper).toMatchSnapshot();
});


test('should call onAction to go to menu', () => {
    const button = wrapper.find('button.menu-button');

    //find one
    expect(button.length).toBe(1);
    //call action
    button.simulate('click');
    //
    expect(onAction).toHaveBeenLastCalledWith('menu');
});


test('should call onAction to reply game', () => {
    const button = wrapper.find('button.replay-button');

    //find one
    expect(button.length).toBe(1);
    //call action
    button.simulate('click');
    //
    expect(onAction).toHaveBeenLastCalledWith('replay');
});
