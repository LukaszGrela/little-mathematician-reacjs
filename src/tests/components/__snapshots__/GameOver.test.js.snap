// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GameOver correctly 1`] = `
<div
  className="game-over"
>
  <div
    className="title"
  >
    <span
      className="p1"
    >
      Game 
    </span>
    <span
      className="p2"
    >
      Over
    </span>
    <span
      className="p3"
    >
      !
    </span>
  </div>
  <div
    className="content"
  >
    You have resolved 
    10
     equation
    s
     correctly.
  </div>
  <div
    className="footer"
  >
    <button
      className="replay-button"
      onClick={[Function]}
    >
      <IconReplay />
    </button>
    <button
      className="menu-button"
      onClick={[Function]}
    >
      <IconMenu />
    </button>
  </div>
</div>
`;

exports[`should render GameOver correctly with 1 correct answer 1`] = `
<div
  className="game-over"
>
  <div
    className="title"
  >
    <span
      className="p1"
    >
      Game 
    </span>
    <span
      className="p2"
    >
      Over
    </span>
    <span
      className="p3"
    >
      !
    </span>
  </div>
  <div
    className="content"
  >
    You have resolved 
    1
     equation
     correctly.
  </div>
  <div
    className="footer"
  >
    <button
      className="replay-button"
      onClick={[Function]}
    >
      <IconReplay />
    </button>
    <button
      className="menu-button"
      onClick={[Function]}
    >
      <IconMenu />
    </button>
  </div>
</div>
`;
